gulpfile.js

```diff
@@ -474,42 +586,58 @@ gulp.task('images:printpdf', function (done) {
-
-    console.log('Processing print-PDF images from ' + paths.img.source);
-    if (fileExists.sync('_tools/profiles/' + printPDFColorProfile)) {
+    var outputFormat = 'print-pdf';
+    var colorProfile = 'PSOcoated_v3.icc';
+    var colorSpace = 'cmyk';
+    var colorProfileGrayscale = defaultColorProfileGrayscale;
+    var colorSpaceGrayscale = defaultColorSpaceGrayscale;
+
+    console.log('Processing ' + outputFormat + ' images from ' + paths.img.source);
+    if (fileExists.sync('_tools/profiles/' + colorProfile)) {
         gulp.src(paths.img.source + '*.{' + filetypes + '}',
                 {ignore: paths.ignore.printpdf})
             .pipe(newer(paths.img.printpdf))
-            .pipe(debug({title: 'Creating print-PDF version of '}))
+
+            // --------------------------------------------------------------
+            // Same for all bitmap image tasks
+            .pipe(debug({title: 'Creating ' + outputFormat + ' version of '}))
             .pipe(gm(function (gmfile) {
 
-                // Check for grayscale
-                var thisColorProfile = printPDFColorProfile; // set default/fallback
-                var thisColorSpace = printPDFColorSpace; // set default/fallback
-                var thisFilename = gmfile.source.split('\/').pop(); // for unix slashes
-                thisFilename = thisFilename.split('\\').pop(); // for windows backslashes
-
-                // Look up image colour settings
-                imageSettings.forEach(function (image) {
-                    if (image.file === thisFilename || image.file == "all") {
-                        if (image['print-pdf'].colorspace === 'gray') {
-                            thisColorProfile = printPDFColorProfileGrayscale;
-                            thisColorSpace = printPDFColorSpaceGrayscale;
-                        }
-                    }
-                });
+                // Get file details
+                var filename = getFilenameFromPath(gmfile.source);
+
+                // Check if we should modify this image
+                var modifyImage = modifyImageCheck(filename, outputFormat);
+
+                // Reset defaults (in case previous image in stream
+                // set these values to something else)
+                var thisColorSpace = colorSpace;
+                var thisColorProfile = colorProfile;
+
+                // Override
+                var colorSettings = lookupColorSettings(gmfile, colorProfile, colorSpace,
+                        colorProfileGrayscale, colorSpaceGrayscale, outputFormat);
+                thisColorSpace = colorSettings.colorSpace;
+                thisColorProfile = colorSettings.colorProfile;
+
+                if (modifyImage) {
+                    return gmfile
+                    .profile('_tools/profiles/' + thisColorProfile)
+                    .colorspace(thisColorSpace);
+                } else {
+                    return gmfile
+                    .define('jpeg:preserve-settings')
```

Comment:
Does this also work for .png?

Reply:
Good question. I've just tested and, essentially, yes. My understanding is that the `jpeg:preserve-settings` flag doesn't have any effect on the .png, but the overall effect is the same: .pngs set to `modify: no` are not changed.

Incidentally, the file size does change slightly, but when I run the images though an image diff there are no visul differences. I suspect that GM still changes metadata in the file, which affects file size but not rendering. Which is fine for our purposes.